modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIgniteJoin.java

```diff
@@ -304,9 +303,10 @@ protected boolean projectsLeft(RelCollation collation) {
         ImmutableIntList sourceKeys = left2Right ? joinInfo.leftKeys : joinInfo.rightKeys;
         ImmutableIntList targetKeys = left2Right ? joinInfo.rightKeys : joinInfo.leftKeys;
 
-        Map<Integer, Integer> keyMap = new HashMap<>();
+        Int2IntOpenHashMap keyMap = new Int2IntOpenHashMap(joinInfo.leftKeys.size());
+
```

Comment:
This change make no sense, a Mappings.target() will box, then unbox int values, in anyway.

Reply:
Agree, reverted it.